OZprivate/rawJS/OZTreeModule/src/position_helper.js

```diff
@@ -267,67 +270,67 @@ function perform_actual_fly(controller, into_node, speed=1, accel_type="linear",
  * @param {func} abrupt_func is optional, and gives a function to call when fly is abrupted
  */
 function perform_fly_b2(controller, into_node, speed, accel_type, finalize_func, abrupt_func) {
-  function pan_proportion(step, total) {
-    var x = step / total,
-        out = (Math.sin((x+0.25) * Math.PI*2) + 1) / 2;
 
-    return accel_type === 'accel' ? (x > 0.5 ? out : 0)
-         : accel_type === 'decel' ? 1 - (x < 0.5 ? out : 0)
-         : x;
+  const time_proportion = (performance.now() - fly_start_time) / 1000 / fly_duration_s;
+  const clamped_time_proportion = Math.max(0, Math.min(time_proportion, 1));
+
+  function pan_proportion(time_proportion) {
+    var out = (Math.sin((time_proportion+0.25) * Math.PI*2) + 1) / 2;
+
+    return accel_type === 'accel' ? (time_proportion > 0.5 ? out : 0)
+         : accel_type === 'decel' ? 1 - (time_proportion < 0.5 ? out : 0)
+         : time_proportion;
   }
-  function zoom_proportion(step, total) {
-    var x = step / total,
-        out = (Math.sin((x+1.5) * Math.PI) + 1) / 2;
+  function zoom_proportion(time_proportion) {
+    var out = (Math.sin((time_proportion+1.5) * Math.PI) + 1) / 2;
 
     return accel_type === 'accel' ? out
          : accel_type === 'decel' ? out
-         : x;
+         : time_proportion;
   }
 
   if (more_flying_needed) {
     //need to reanchor, this sometimes causes jerkiness
     //also we may not know how many steps we will need to take
     // NB: Approximate accel/decel by not panning at all, let the final non-reanchor step handle it
-    pan_zoom(accel_type === 'accel' || accel_type === 'decel'? 0 : 1/num_intro_steps, 1/num_intro_steps);
+    pan_zoom(accel_type === 'accel' || accel_type === 'decel'? 0 : clamped_time_proportion, clamped_time_proportion);
```

Comment:
At the moment, we do this thing of going back to `perform_actual_fly` every frame when we're going a long way. This means that we'd ideally want to be more careful about capturing all non-trivial work between when we start the timer (fly_start_time) and when we measure the time proportion. So ideally we'd get all the re_calc/reanchor stuff inside that interval.

Buuut I wasn't totally sure why we do the re-anchoring this frequently anyway. Since we're limiting the r_mult range to 1M either way might it not make sense to just re-anchor at the end of this planned segment of the flight rather than every frame?

Reply:
Yeah that makes sense. Perhaps a sensible scheme would be to chop the interval in half if it's greater than the interval